src/libraries/System.Net.Http/src/System/Net/Http/HttpMethod.SocketsHttpHandler.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Net.Http.HPack;
+using System.Net.Http.QPack;
+using System.Text;
+
+namespace System.Net.Http
+{
+    public partial class HttpMethod
+    {
+        private byte[]? _http1EncodedBytes;
+        private byte[]? _http2EncodedBytes;
+        private byte[]? _http3EncodedBytes;
+        private int _http3Index;
```

Comment:
I assume we believe it's very rare to create a new HttpMethod instance? If we did, this would be penalizing it.

Reply:
I assume most code uses the static instances like `HttpMethod.Get`?

Reply:
You would think, but https://grep.app/search?q=new%20HttpMethod%28&filter[lang][0]=C%23
Seems like we should be fixing the Azure SDK generator tools.

Reply:
I did see the Azure code before putting up the PR, and it looks like they've updated their generator to [reuse objects](https://github.com/Azure/autorest.csharp/blob/23dcd81c26074c182961ae4a10319915ec4026ff/src/AutoRest.CSharp/Common/Input/AzureApiTypes.cs#L53) since, but haven't regenerated the sources.

As long as it's a known method that something else in the process used the singleton for, you'd still avoid the allocation (we check the cached bytes on the known method first).

Reply:
It's simple enough to avoid it even if the user is intent on never using the singleton though :)
Pushed that change now